src/layouts/Authenticated.vue

```diff
@@ -446,29 +447,88 @@ export default defineComponent({
       }
     });
 
+    /**
+     * Order of priority of incident id (descending):
+     * - Route path parameter (incidents/123/work)
+     * - User states current incident id
+     * - Most recent incident they have access too
+     */
+    async function resolveInitialIncidentId() {
```

Comment:
1.) you should move all this logic out to a dedicated hook (iirc, I had a `useCurrentIncident` hook in v3)
2.) instead of dereferencing `currentUser` use it as intended, which is reactivity. `recentIncidentIdFromState` should be computed from this etc.